assets/js/modules/analytics-4/datastore/custom-dimensions.js

```diff
@@ -204,42 +222,97 @@ const baseActions = {
 
 		// Sync available custom dimensions.
 		if ( missingCustomDimensions.length > 0 ) {
-			yield baseActions.syncAvailableCustomDimensions( propertyID );
+			yield baseActions.syncAvailableCustomDimensions();
 		}
+
+		yield {
+			type: SET_CUSTOM_DIMENSIONS_BEING_CREATED,
+			payload: { customDimensions: [] },
+		};
 	},
 
 	/**
 	 * Syncs available custom dimensions in the settings.
 	 *
 	 * @since n.e.x.t
 	 *
-	 * @param {string} propertyID GA4 property ID.
 	 * @return {Array<string>} Available custom dimensions.
 	 */
-	syncAvailableCustomDimensions: createValidatedAction(
-		( propertyID ) => {
-			invariant(
-				isValidPropertyID( propertyID ),
-				'A valid GA4 propertyID is required.'
+	*syncAvailableCustomDimensions() {
+		const registry = yield Data.commonActions.getRegistry();
+
+		const propertyID = registry
+			.select( MODULES_ANALYTICS_4 )
+			.getPropertyID();
+
+		yield {
+			type: SET_AVAILABLE_CUSTOM_DIMENSIONS_SYNCING_ACTION,
+			payload: { isSyncing: true },
+		};
+
+		const { response, error } =
+			yield fetchSyncAvailableCustomDimensionsStore.actions.fetchSyncAvailableCustomDimensions(
+				propertyID
 			);
-		},
-		function* ( propertyID ) {
-			const registry = yield Data.commonActions.getRegistry();
 
-			const { response, error } =
-				yield fetchSyncAvailableCustomDimensionsStore.actions.fetchSyncAvailableCustomDimensions(
-					propertyID
-				);
+		if ( response ) {
+			yield registry
+				.dispatch( MODULES_ANALYTICS_4 )
+				.setAvailableCustomDimensions( response );
+		}
 
-			if ( response ) {
-				registry
-					.dispatch( MODULES_ANALYTICS_4 )
-					.setAvailableCustomDimensions( response );
-			}
+		yield {
+			type: SET_AVAILABLE_CUSTOM_DIMENSIONS_SYNCING_ACTION,
+			payload: { isSyncing: false },
+		};
+
+		return { response, error };
+	},
+
+	/**
+	 * Sets custom dimension creation error in the store.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} error           The error object.
+	 * @param {string} customDimension Custom dimension to set creation error for.
+	 */
+	*receiveCreateCustomDimensionError( error, customDimension ) {
```

Comment:
Hi @eugene-manuilov.

I actually only added this action for easier datastore mocking in stories and tests (for now). It is not being used elsewhere.

Without an action like this, we'll have to export `possibleCustomDimensions` (as a constant, for example) so that we can use it with `receiveError`.

Reply:
Let's make this action to be a utility function then and have the same approach as we use for `provide*` helpers. So we will have something like this: `provideCustomDimensionsError( registry, { customDimension, error } )`.

> Without an action like this, we'll have to export `possibleCustomDimensions` (as a constant, for example)

Fine with me.